File: Pascal/Niveau 1/2 – Répétitions d'instructions/Le Grand Événement.pas
program Solution;
uses Robot;
var
   i: LongInt;
   j: LongInt;
begin
   haut();
   for i := 1 to 4 do
   begin
      for j := 1 to 8 do
      begin
         haut;
      end;
      droite;
      for j := 1 to 8 do
      begin
         bas;
      end;
      droite;
   end;
   for i := 1 to 8 do
   begin
      haut;
   end;
   droite;
   for i := 1 to 9 do
   begin
      bas;
   end;
   for i := 1 to 9 do
   begin
      gauche;
   end;
end.

File: OCaml/Niveau 1/2 – Répétitions d'instructions/Le Grand Événement.ml
open Robot;;

haut();
for i = 1 to 4 do
  for j = 1 to 8 do
    haut();
  done;
  droite();
  for j = 1 to 8 do
    bas();
  done;
  droite();
done;
for loop = 1 to 8 do
  haut();
done;
droite();
for loop = 1 to 9 do
  bas();
done;
for loop = 1 to 9 do
  gauche();
done;